src/metaprob/examples/variational_on_gaussians.clj

```diff
@@ -0,0 +1,202 @@
+(ns metaprob.examples.variational-on-gaussians
+  (:require [metaprob.inference :as infer]
+            [clojure.pprint :refer [pprint]]
+            [metaprob.distributions :as dist]
+            [metaprob.prelude :as mp]
+            [metaprob.trace :as trace]
+            [metaprob.generative-functions :refer [at gen let-traced]]
+            [metaprob.autodiff :as ad]))
+
+(def beta-binomial
+  (gen [n]
+    (let-traced [p1 (dist/beta 1 1)
+                 p2 (dist/beta 1 1)]
+      (doall (map #(at % dist/flip (if (at `("which-p" ~%) dist/flip 0.5) p1 p2)) (range n))))))
+
+(defn make-observation-trace [l]
+  (first (reduce (fn [[tr i] x] [(trace/trace-set-value tr i x) (inc i)]) [{} 0] l)))
+
+(defn initial-trace [l]
+  (trace/trace-clear-subtrace
+    ((mp/infer-and-score :procedure beta-binomial
+                         :inputs [(count l)]
+                         :observation-trace (make-observation-trace l)) 1)
+    "which-p"))
+
+
+(defn map-optimize-demo []
+  (pprint
+    (let [tr
+          (time
+            (last
+              (take 200
+                (iterate
+                  (infer/map-optimize-step
+                    :model beta-binomial
+                    :inputs [200]
+                    :step-size 0.0001
+                    :addresses '("p1" "p2"))
+                  (initial-trace (vec (concat (repeat 100 true) (repeat 100 false))))))))]
+  [(trace/trace-value tr "p1") (trace/trace-value tr "p2")])))
+
+
+; Model
+(def normal-normal
+  (gen []
+    (let-traced [x (dist/gaussian 0 1)
+                 y (dist/gaussian x 1)]
+      y)))
+
+; Guide
+(defn normal-normal-predicter
```

Comment:
Nit: do you feel strongly about "predicter" vs. "predictor"?

Reply:
I do not :)

Reply:
Thanks :)